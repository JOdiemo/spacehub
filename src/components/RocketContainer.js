import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getRockets } from '../redux/Rockets/RocketsSlice';
import RocketPopulate from './RocketPopulate';
import './rockets.css';

function RocketContainer() {
  const rockets = useSelector((state) => state.rockets);
  const dispatch = useDispatch();

  useEffect(() => {
    if (rockets.length === 0) {
      dispatch(getRockets());
    }
  }, [dispatch, rockets.length]);
  return (
    <div className="rocketItems">
      {rockets.map((rocket) => (
        <RocketPopulate
          key={rocket.id}
          id={rocket.id}
          reserved={rocket?.reserved}
          flickr_images={rocket.flickr_images[0]}
          name={rocket.name}
          description={rocket.description}
        />
      ))}
    </div>
  );
}

export default RocketContainer;
