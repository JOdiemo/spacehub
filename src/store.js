import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './redux/Rockets/RocketsSlice';

const store = configureStore({
  reducer: {
    rocket: rocketReducer,
  },
});

export default store;
