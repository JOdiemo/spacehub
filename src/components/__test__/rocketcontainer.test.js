import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../store';
import RocketContainer from '../RocketContainer';

describe('Navbar component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <RocketContainer />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
