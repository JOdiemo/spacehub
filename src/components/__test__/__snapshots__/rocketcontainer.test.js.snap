// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component matches the snapshot 1`] = `
<div
  className="rocketItems"
/>
`;
